In [1]:
%load_ext autoreload
%autoreload 2

In [8]:
import sys
sys.path.append('../src')  #path to src

import pandas as pd
import os
from data_loader import *

#  📊 Raw Data Exploration | Meat Consumption Project

**Description:** 

In this notebook, we load and explore the raw datasets related to global meat consumption. The goal is to understand the structure, completeness, and key variables of each dataset. 



--
**Structure:**
- [Overview of Raw Datasets](#overview-of-raw-datasets)
- [FAO Dataset](#fao-dataset)
- [GDP Dataset](#gdp-dataset)
- [Urban Dataset](#urban-dataset)
- [Edudcation Dataset](#education-dataset)
- [Enviroment Dataset](#enviroment-dataset)
- [Production Dataset](#meat-production-dataset)
- [Command summary throughout this notebook](#command-summary-thorugout-this-notebook)

--

## Overview of Raw Datasets




This project integrates multiple datasets from different sources to explore global meat consumption patterns. Below is a summary of each dataset and its primary content:

- **FAOSTAT – Meat Consumption per Capita**  
  Annual per capita meat consumption (kg/year) for ~190 countries, disaggregated by meat type.  
  Time span: 2010–2022

- **World Bank – GDP per Capita**  
  Economic indicator representing average income per person.  
  Time span: 1960–2024 (used: 2010–2022)

- **World Bank – Urban Population (%)**  
  Share of the population living in urban areas, by country.  
  Time span: 1960–2024 (used: 2010–2022)

- **OWID – Basic Education (%)**  
  Share of population with at least basic education.  
  Limited to 25 countries.  
  Time span: 1820–2100

- **OWID – Environmental Impact of Food**  
  Greenhouse gas (GHG) emissions per kg of food product.  
  Used for contextual environmental analysis.  
  Year: 2010

- **OWID – Global Meat Production**  
  Total meat production by country and year (not per capita).  
  Time span: 1961–2023  
  Optional, used for additional visualization or context.

Each dataset contributes to understanding how socioeconomic and environmental factors correlate with meat consumption across countries and time.

--

## Exploratory Data Analysis

In [9]:
# Calling the files
dfs = load_raw_data()

### FAO Dataset

#### Initial Exploration

In [28]:
# First we simplify the name of the frame we will explore. 
faostat = dfs['faostat']

In [29]:
# File header structure
faostat.head(3)

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code (FBS),Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
0,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2010,2010,kg/cap,4.74,E,Estimated value,NaN
1,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2011,2011,kg/cap,4.80,E,Estimated value,NaN
2,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2012,2012,kg/cap,4.40,E,Estimated value,NaN


In [30]:
# file tail structure
faostat.tail(3)

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code (FBS),Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
11862,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2020,2020,kg/cap,2.35,E,Estimated value,NaN
11863,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2021,2021,kg/cap,2.35,E,Estimated value,NaN
11864,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2022,2022,kg/cap,2.35,E,Estimated value,NaN


In [22]:
faostat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11865 entries, 0 to 11864
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Domain Code       11865 non-null  object 
 1   Domain            11865 non-null  object 
 2   Area Code (M49)   11865 non-null  int64  
 3   Area              11865 non-null  object 
 4   Element Code      11865 non-null  int64  
 5   Element           11865 non-null  object 
 6   Item Code (FBS)   11865 non-null  object 
 7   Item              11865 non-null  object 
 8   Year Code         11865 non-null  int64  
 9   Year              11865 non-null  int64  
 10  Unit              11865 non-null  object 
 11  Value             11865 non-null  float64
 12  Flag              11865 non-null  object 
 13  Flag Description  11865 non-null  object 
 14  Note              0 non-null      float64
dtypes: float64(2), int64(4), object(9)
memory usage: 1.4+ MB


In [23]:
faostat.isnull().sum()

Domain Code             0
Domain                  0
Area Code (M49)         0
Area                    0
Element Code            0
Element                 0
Item Code (FBS)         0
Item                    0
Year Code               0
Year                    0
Unit                    0
Value                   0
Flag                    0
Flag Description        0
Note                11865
dtype: int64

In [24]:
print(f"Dataset shape: {faostat.shape[0]} rows × {faostat.shape[1]} columns")

Dataset shape: 11865 rows × 15 columns


In [25]:
# General exploration, number of data and features
print('Unic values in  columns')
print(faostat['Domain'].unique())
print(faostat['Item Code (FBS)'].unique())
print(faostat['Item'].unique())
print('Countries:', faostat['Area'].nunique())
print('Year range:',faostat['Year'].min(), '-' , faostat['Year'].max())
print('Suspicious values',faostat['Flag'].value_counts())

Unic values in  columns
['Food Balances (2010-)']
['S2731' 'S2732' 'S2733' 'S2734' 'S2735']
['Bovine Meat' 'Mutton & Goat Meat' 'Pigmeat' 'Poultry Meat' 'Meat, Other']
Countries: 190
Year range: 2010 - 2022
Suspicious values Flag
E    11865
Name: count, dtype: int64


In [26]:
faostat[['Item Code (FBS)', 'Item']].drop_duplicates().sort_values('Item Code (FBS)')


,Item Code (FBS),Item
0,S2731,Bovine Meat
13,S2732,Mutton & Goat Meat
26,S2733,Pigmeat
35,S2734,Poultry Meat
48,S2735,"Meat, Other"


In [27]:
faostat[['Area', 'Area Code (M49)']].drop_duplicates().sort_values('Area Code (M49)')

,Area,Area Code (M49)
0,Afghanistan,4
61,Albania,8
126,Algeria,12
191,Angola,24
256,Antigua and Barbuda,28
...,...,...
11410,Uzbekistan,860
11540,Venezuela (Bolivarian Republic of),862
9173,Samoa,882
11670,Yemen,887


In [31]:
faostat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11865 entries, 0 to 11864
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Domain Code       11865 non-null  object 
 1   Domain            11865 non-null  object 
 2   Area Code (M49)   11865 non-null  int64  
 3   Area              11865 non-null  object 
 4   Element Code      11865 non-null  int64  
 5   Element           11865 non-null  object 
 6   Item Code (FBS)   11865 non-null  object 
 7   Item              11865 non-null  object 
 8   Year Code         11865 non-null  int64  
 9   Year              11865 non-null  int64  
 10  Unit              11865 non-null  object 
 11  Value             11865 non-null  float64
 12  Flag              11865 non-null  object 
 13  Flag Description  11865 non-null  object 
 14  Note              0 non-null      float64
dtypes: float64(2), int64(4), object(9)
memory usage: 1.4+ MB


#### Summary of FAOSTAT Dataset Initial Exploration
--
**Dataset size**
- Rows: 11865
- columns: 15

**Key feature/columns**
- `Area`: Country (object)
- `Item`: Meet type (object)
  - Related `Item Code (FBS)`(object) mapping: 
    - `S2731`: Bovine Meat
    - `S2732`: Mutton & Goat Meat
    - `S2733`: Pigmeat
    - `S2734`: Poultry Meat
    - `S2735`: Meat, Other
- `Year`: Measure  year (int64)
- `Value`: Consumption (kg/capita/year) (float64)

**Periodcoverage**
- Years: 2010 - 2022

**Geography coverage**
- number of countries: 190

**Missing names**
- `Notes` is the only column with `NaN` (irrelevant)

**Relevant notes**
- Some rows contain `Item = Meat, Other`: This could be excluded for our analysis
- `Element` = "Fuud Supply quality (kg/capita/yr)" is the relevant input for training.

**Suggested actions for processing**
- Filter by `Element`
- Exclude `Item = "Meat, Other"`
- Rename key columns

### GDP Dataset

#### Initial Exploration

In [13]:
# First we simplify the name of the frame we will explore. 
gdp = dfs['gdp']

In [14]:
# File header structure
gdp.head(3)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
0,Aruba,ABW,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,27441.529662,28440.051964,30082.127645,31096.205074,22855.932320,27200.061079,30559.533535,33984.790620,NaN,NaN
1,Africa Eastern and Southern,AFE,GDP per capita (current US$),NY.GDP.PCAP.CD,186.121835,186.941781,197.402402,225.440494,208.999748,226.876513,...,1329.807285,1520.212231,1538.901679,1493.817938,1344.103210,1522.393346,1628.318944,1568.159891,1673.841139,NaN
2,Afghanistan,AFG,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,522.082216,525.469771,491.337221,496.602504,510.787063,356.496214,357.261153,413.757895,NaN,NaN


In [15]:
gdp.tail(3)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
263,South Africa,ZAF,GDP per capita (current US$),NY.GDP.PCAP.CD,532.147504,545.657512,563.423009,604.536855,645.873376,684.621228,...,5651.205852,6618.335083,6914.178032,6533.711210,5580.603831,6843.399419,6523.410978,6022.542542,6253.371582,NaN
264,Zambia,ZMB,GDP per capita (current US$),NY.GDP.PCAP.CD,221.559849,209.693206,202.281031,203.219451,229.979246,287.425476,...,1239.085279,1483.465773,1463.899979,1258.986198,951.644317,1127.160779,1447.123101,1330.727806,1235.084665,NaN
265,Zimbabwe,ZWE,GDP per capita (current US$),NY.GDP.PCAP.CD,276.419784,279.016489,275.545608,277.005701,281.744539,294.145359,...,1407.420964,3448.086991,2271.852504,1683.913136,1730.453910,1724.387271,2040.546587,2156.034093,2656.409377,NaN


In [16]:
gdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 70 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            151 non-null    float64
 5   1961            154 non-null    float64
 6   1962            156 non-null    float64
 7   1963            156 non-null    float64
 8   1964            156 non-null    float64
 9   1965            162 non-null    float64
 10  1966            163 non-null    float64
 11  1967            167 non-null    float64
 12  1968            168 non-null    float64
 13  1969            168 non-null    float64
 14  1970            190 non-null    float64
 15  1971            191 non-null    float64
 16  1972            191 non-null    float64
 17  1973            191 non-null    flo

In [17]:
gdp.isnull().sum()

Country Name        0
Country Code        0
Indicator Name      0
Indicator Code      0
1960              115
                 ... 
2021                8
2022                9
2023               17
2024               34
Unnamed: 69       266
Length: 70, dtype: int64

No Nulls in `Country Name`, `Country Code`, `Indicator Name`, `Indicator Code` but many Nulls from `1960` - `2024`

In [94]:
# General exploration, number of data and features
print(f"Dataset shape: {gdp.shape[0]} rows × {gdp.shape[1]} columns")

print(f'columns: {gdp.columns}')

Dataset shape: 262 rows × 69 columns
columns: Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022',
       '2023', '2024'],
      dtype='object')


In [ ]:
# Dropping named 69
gdp = gdp.drop(columns=['Unnamed: 69'])


In [20]:

year_cols = [col for col in gdp.columns if col.isdigit()]  # columns in years
gdp = gdp.dropna(subset=year_cols, how='all')

In [21]:
# General exploration, number of data and features
print('(rows, columns):',gdp.shape)

print('columns:', gdp.columns)

(rows, columns): (262, 69)
columns: Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022',
       '2023', '2024'],
      dtype='object')


In [22]:
gdp.isnull().sum()

Country Name        0
Country Code        0
Indicator Name      0
Indicator Code      0
1960              111
                 ... 
2020                4
2021                4
2022                5
2023               13
2024               30
Length: 69, dtype: int64

In [24]:

print('Number of countries:', gdp['Country Name'].nunique())
print('(rows,columns)', gdp.shape)


Number of countries: 262
(rows,columns) (262, 69)


#### Summary of GDP dataset

--

**Datasert size**
- rows: 266
- columns: 70

**Geography coverage**
- Number of countries: 266

**Period coverage**
- Years: 1960 - 2024

**Key features**
- `Country Name` (object)
- Year columns: from `2010` to `2022`

**Suggesting actions for  processing**
- Dropp column `Unnamed 69`: Nan values
- There are still Null values in the final table but, we will handle this later  when processing data
- Keep only relevant columns: from `2010` to `2022`

### Urban Dataset

#### Initial Exploration

In [25]:
urban = dfs['urban']

In [28]:
urban.head(1)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
0,Aruba,ABW,Urban population (% of total population),SP.URB.TOTL.IN.ZS,50.776,50.761,50.746,50.73,50.715,50.7,...,43.192,43.293,43.411,43.546,43.697,43.866,44.052,44.254,44.474,NaN


In [29]:
urban.tail(3)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
263,South Africa,ZAF,Urban population (% of total population),SP.URB.TOTL.IN.ZS,46.619,46.793,46.906,47.020,47.134,47.248,...,65.341,65.850,66.355,66.856,67.354,67.847,68.335,68.819,69.298,NaN
264,Zambia,ZMB,Urban population (% of total population),SP.URB.TOTL.IN.ZS,18.145,18.951,19.785,20.712,22.015,23.372,...,42.438,42.976,43.521,44.072,44.629,45.192,45.761,46.335,46.914,NaN
265,Zimbabwe,ZWE,Urban population (% of total population),SP.URB.TOTL.IN.ZS,12.608,12.821,13.082,13.578,14.092,14.620,...,32.296,32.237,32.209,32.210,32.242,32.303,32.395,32.517,32.670,NaN


In [30]:
urban.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 70 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            263 non-null    float64
 5   1961            263 non-null    float64
 6   1962            263 non-null    float64
 7   1963            263 non-null    float64
 8   1964            263 non-null    float64
 9   1965            263 non-null    float64
 10  1966            263 non-null    float64
 11  1967            263 non-null    float64
 12  1968            263 non-null    float64
 13  1969            263 non-null    float64
 14  1970            263 non-null    float64
 15  1971            263 non-null    float64
 16  1972            263 non-null    float64
 17  1973            263 non-null    flo

In [31]:
urban.isnull().sum()

Country Name        0
Country Code        0
Indicator Name      0
Indicator Code      0
1960                3
                 ... 
2021                3
2022                3
2023                3
2024                3
Unnamed: 69       266
Length: 70, dtype: int64

In [32]:
# Dropping named
urban = urban.drop(columns=['Unnamed: 69'])

In [95]:
# General exploration, number of data and features
print(f"Dataset shape: {urban.shape[0]} rows × {urban.shape[1]} columns")

print(f'columns: {urban.columns}')

Dataset shape: 266 rows × 69 columns
columns: Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022',
       '2023', '2024'],
      dtype='object')


In [39]:
urban['Country Name'].nunique()

266

#### Summary of Urban dataset

--

**Datasert size**
- rows: 266
- columns: 70

**Geography coverage**
- Number of countries: 266

**Period coverage**
- Years: 1960 - 2024

**Key features**
- `Country Name` (object)
- Year columns: from `2010` to `2022`

**Suggesting actions for  processing**
- Dropp column `Unnamed 69`: contains only NaN values
- There are still Null values in the final table but, we will handle this later  when processing data
- Keep only relevant columns: from `2010` to `2022`

### Education Dataset

#### Initial Exploration

In [40]:
education = dfs['education']

In [41]:
education.shape

(815, 5)

In [ ]:
education.head(3)

NameError: name 'education' is not defined

In [43]:
education.tail(3)

,Entity,Code,Year,Share of population with no education,Share of population with at least some basic education
812,World,OWID_WRL,2090,2.354454,97.645546
813,World,OWID_WRL,2095,2.051334,97.948670
814,World,OWID_WRL,2100,1.792332,98.207664


In [44]:
education.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 815 entries, 0 to 814
Data columns (total 5 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   Entity                                                  815 non-null    object 
 1   Code                                                    815 non-null    object 
 2   Year                                                    815 non-null    int64  
 3   Share of population with no education                   815 non-null    float64
 4   Share of population with at least some basic education  815 non-null    float64
dtypes: float64(2), int64(1), object(2)
memory usage: 32.0+ KB


In [46]:
education.columns

Index(['Entity', 'Code', 'Year', 'Share of population with no education',
       'Share of population with at least some basic education'],
      dtype='object')

In [47]:
education.isnull().sum()

Entity                                                    0
Code                                                      0
Year                                                      0
Share of population with no education                     0
Share of population with at least some basic education    0
dtype: int64

In [49]:
education['Entity'].nunique()

26

In [53]:
print(education['Year'].min(),education['Year'].max())

1820 2100


#### Summary of Education dataset
--
**Size of dataset**
- rows: 815
- columns: 5

**Geography coverage**
- Countries: 25
**Period coverage**
- Years: 1820 - 2100

**Key Features**
- `Entity`: Country (object)
- `Year` (int64)
- `Share of population with no education ` (float64)
- `Share of population with at least some basic education` (float64)


### Enviroment Dataset

#### Initial Exploration

In [55]:
envt = dfs['enviroment']

In [56]:
envt.shape

(38, 3)

In [57]:
envt.head(3)

,Entity,Year,"GHG emissions per kilogram (Poore & Nemecek, 2018)"
0,Apples,2010,0.43
1,Bananas,2010,0.86
2,Barley,2010,1.18


In [58]:
envt.tail(3)

,Entity,Year,"GHG emissions per kilogram (Poore & Nemecek, 2018)"
35,Tomatoes,2010,2.09
36,Wheat & Rye,2010,1.57
37,Wine,2010,1.79


In [59]:
envt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38 entries, 0 to 37
Data columns (total 3 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   Entity                                              38 non-null     object 
 1   Year                                                38 non-null     int64  
 2   GHG emissions per kilogram (Poore & Nemecek, 2018)  38 non-null     float64
dtypes: float64(1), int64(1), object(1)
memory usage: 1.0+ KB


In [60]:
envt.isnull().sum()

Entity                                                0
Year                                                  0
GHG emissions per kilogram (Poore & Nemecek, 2018)    0
dtype: int64

In [61]:
envt['Entity'].nunique()

38

In [64]:
envt['Year'].max()

np.int64(2010)

#### Summary of Enviroment dataset
--
**Size of dataset**
- rows: 38
- columns: 3

**Product coverage**
- Product: 38
**Period coverage**
- Years: 2010

**Key Features**
- `Entity`: Product
- `Year`
- `GHG emissions per kilogram (Poore & Nemecek, 2018)`


### Meat production Dataset

#### Initial Exploration

In [66]:
meatp = dfs['production']

In [68]:
meatp.shape

(14614, 4)

In [69]:
meatp.head(3)

,Entity,Code,Year,"Meat, total | 00001765 || Production | 005510 || tonnes"
0,Afghanistan,AFG,1961,129420.00
1,Afghanistan,AFG,1962,132205.73
2,Afghanistan,AFG,1963,138971.36


In [70]:
meatp.tail(3)

,Entity,Code,Year,"Meat, total | 00001765 || Production | 005510 || tonnes"
14611,Zimbabwe,ZWE,2021,895893.50
14612,Zimbabwe,ZWE,2022,926426.20
14613,Zimbabwe,ZWE,2023,933320.94


In [77]:
meatp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14614 entries, 0 to 14613
Data columns (total 4 columns):
 #   Column                                                   Non-Null Count  Dtype  
---  ------                                                   --------------  -----  
 0   Entity                                                   14614 non-null  object 
 1   Code                                                     11909 non-null  object 
 2   Year                                                     14614 non-null  int64  
 3   Meat, total | 00001765 || Production | 005510 || tonnes  14614 non-null  float64
dtypes: float64(1), int64(1), object(2)
memory usage: 456.8+ KB


In [78]:
meatp.isnull().sum()

Entity                                                        0
Code                                                       2705
Year                                                          0
Meat, total | 00001765 || Production | 005510 || tonnes       0
dtype: int64

In [79]:
meatp['Year'].min()

np.int64(1961)

In [81]:
meatp['Year'].max()

np.int64(2023)

In [82]:
meatp['Entity'].nunique()

254

#### Summary of production dataset
--
**Size of dataset**
- rows: 14614
- columns: 4

**Geography coverage**
- Contries: 255
**Period coverage**
- Years: 1961 - 2023

**Key Features**
- `Entity`: Country
- `Year`
- `Meat, total | 00001765 || Production | 005510 || tonnes`

**Useful notes**
- Last colummn is production in tonnes, might be useful to convert into kg
- We canignore `Code` column


## Command summary througout this notebook




Using `df` as the general name for the dataset, the following commands will be used throughout the notebook:

- `df.shape`  
  Returns the dimensions of the dataset *(rows, columns)*.

- `df.head(3)`  
  Displays the first 3 rows to preview the structure of the dataset.

- `df.tail(3)`  
  Displays the last 3 rows to inspect the dataset's end.

- `df.info()`  
  Provides a summary of the dataset, including column types and non-null counts.

- `df.columns`  
  Displays the name of each column in the dataset.

- `df.isnull().sum()`  
  Returns the number of missing (`Null`) values per column.

- `df['feature1'].nunique()`  
  Returns the number of unique values in the column `feature1`.

- `df['feature1'].min()`, `df['feature1'].max()`  
  Show the minimum and maximum values in `feature1`, i.e., its range.

- `df[['feature1', 'feature2']].drop_duplicates().sort_values('feature2')`  
  Displays unique combinations of `feature1` and `feature2`, sorted by `feature2`.